FUSETOOLS-2283: New Fuse Integration Project wizard - an ugly rectangle

jpav opened the pull request.

jpav commented:
Jenkins is faling with this, but I have no idea why; apparently something to do with JMX based upon the error messages shown in the console.  Same type of thing happens locally, and it seems pretty consistent.  AFAICT, the actual issue has been fixed properly.  I don't think the test failure is related, but I'm over my head with whatever is happening there.  Can one of you guys more familiar with this take a look?

bfitzpat commented:
Known issue. @lhein  and @apupier have been looking into it for a while and now Nick is helping too I think.

jpav commented:
So it’s okay to be reviewed and checked in then, I assume?

> On Jun 12, 2017, at 4:52 PM, Brian Fitzpatrick <notifications@github.com> wrote:
> 
> Known issue. @lhein <https://github.com/lhein> and @apupier <https://github.com/apupier> have been looking into it for a while and now Nick is helping too I think.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/jbosstools/jbosstools-fuse/pull/981#issuecomment-307942323>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAarXdCeQe336iRvDzZ-w7rNtjyn7uunks5sDbMcgaJpZM4N3fAk>.
>

bfitzpat commented:
Nope. Reviewed-- yes. Merged -- probably not until we get the build/test failures figured out locally and on Jenkins.

lhein commented:
the 2 launch files don't really belong into this PR. Furthermore I'd like to add that the wizard page you reworked will change with the decouple PR  #945. Please make sure that your fix is ported to that PR as well.

apupier commented:
> Furthermore I'd like to add that the wizard page you reworked will change with the decouple PR #945. Please make sure that your fix is ported to that PR as well.

I think that the merge will need to be done in the PR which will be merged the latest. I see no reason to block a merge for synchronization of branch.

lhein commented:
I did not block the PR because of the sync but because of the 2 launch files which don't belong there.

apupier commented:
testPR

apupier commented on editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardRuntimeAndCamelPage.java:
```diff
@@ -75,18 +67,19 @@
 	/* (non-Javadoc)
 	 * @see org.eclipse.jface.dialogs.IDialogPage#createControl(org.eclipse.swt.widgets.Composite)
 	 */
+	@SuppressWarnings("unused")
```
this "SuppressWarnings" is reported as unused so it can be removed

jpav commented:
Apparently my Eclipse prefs were a bit different than yours.  I've changed that and removed the @SuppressWarnings.  Can someone approve this now?

apupier commented:
> Apparently my Eclipse prefs were a bit different than yours. I've changed that and removed the @SuppressWarnings.

it means that i have some Eclipse warnings not activated? Can you tell which warning do you have when you remove this @SuppressWarnings?

apupier commented:
@jpav are you on Mac? Would be nice to provide a screenshot of the new UI with your fix to easily compare with the old one (and to documen tit in the JIRA after)

jpav commented:
I am on a Mac.  I’ll include the screenshot in the JIRA.

> On Jun 28, 2017, at 9:33 AM, Aurélien Pupier <notifications@github.com> wrote:
> 
> @jpav <https://github.com/jpav> are you on Mac? Would be nice to provide a screenshot of the new UI with your fix to easily compare with the old one (and to documen tit in the JIRA after)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/jbosstools/jbosstools-fuse/pull/981#issuecomment-311678452>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAarXSRBIyZCqem9Dx1fTa949UYSxs1gks5sImRDgaJpZM4N3fAk>.
>

jpav commented:
The warning is under Java > Compiler > Error/Warnings > Potential programming problems > Unused object allocation.  I had set to warning.

> On Jun 28, 2017, at 9:32 AM, Aurélien Pupier <notifications@github.com> wrote:
> 
> Apparently my Eclipse prefs were a bit different than yours. I've changed that and removed the @SuppressWarnings <https://github.com/suppresswarnings>.
> 
> it means that i have some Eclipse warnings not activated? Can you tell which warning do you have when you remove this @SuppressWarnings <https://github.com/suppresswarnings>?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/jbosstools/jbosstools-fuse/pull/981#issuecomment-311678153>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAarXd3gtKC-qOZxCTw7_aAqFaLDDLOmks5sImQGgaJpZM4N3fAk>.
>

apupier commented:
> The warning is under Java > Compiler > Error/Warnings > Potential programming problems > Unused object allocation.  I had set to warning.

ok, i see the warning in this case but they seem not very relevant.
SonarLint is not reporting them neither. I think that it is better to ignore these ones. What do you think? Do you have some specific cases in which it is very useful?

jpav commented:
There has been a time or two when this has caught some defunct code, but I’m totally good with this being set to ignore.  I’ve already changed it to that on my side.

> On Jun 28, 2017, at 10:45 AM, Aurélien Pupier <notifications@github.com> wrote:
> 
> The warning is under Java > Compiler > Error/Warnings > Potential programming problems > Unused object allocation. I had set to warning.
> 
> ok, i see the warning in this case but they seem not very relevant.
> SonarLint is not reporting them neither. I think that it is better to ignore these ones. What do you think? Do you have some specific cases in which it is very useful?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/jbosstools/jbosstools-fuse/pull/981#issuecomment-311701355>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAarXdHq0FPkncKrpcnTua3bj3T0dq6Zks5sInUXgaJpZM4N3fAk>.
>
